cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -331,6 +332,42 @@ TEST_F(JITLibTests, ExternalModuleTest) {
   EXPECT_EQ(ptr(), 999);
 }
 
+TEST_F(JITLibTests, TupleTest) {
+  LLVMJITModule module("Test Module");
+
+  JITFunctionPointer function1 = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_Tuple",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "a", .type = JITTypeTag::INT32}}});
+
+  JITValuePointer x_in = function1->createVariable(JITTypeTag::INT32, "x1");
+  JITValuePointer a_in = function1->getArgument(0);
+  JITValuePointer b_in = function1->createConstant(JITTypeTag::INT32, 876);
+
+  JITTuple layer1 = function1->createJITTuple();
+  JITTuple layer2 = function1->createJITTuple();
+  JITTuple layer3 = function1->createJITTuple();
+
+  layer2.append(b_in);
+  layer1.append(x_in);
+  layer1.append(a_in);
+  layer1.insert(layer2, 1);
+  layer1.append(layer3);
+
+  auto& x_out = layer1.getElementAs<JITValue>(0);
+  auto& layer2_out = layer1.getElementAs<JITTuple>(1);
+  auto& a_out = layer1.getElementAs<JITValue>(2);
+  auto& b_out = layer2_out.getElementAs<JITValue>(0);
+
+  x_out = a_out + b_out;
```

Comment:
seems I can't create JITTuple in fun1 and then pass it to func2 outside of func1?

Reply:
you can create JITTuple out of fun1 and then pass it to func1 and func2 ?
Otherwise, JITTuple take over the variables, it will be more complicated to manage?